### PR TITLE
fix: tag selection

### PR DESCRIPTION
**Pull request by Pog3k**

### CODE REVIEW

File: src/pykiso/test_coordinator/test_execution.py [hunk 1]
```diff
@@ -98,18 +98,33 @@ def apply_tag_filter(
...
-        for tag_id, tag_value in usr_tags.items():
+        if test_case.tag is None:
+            return False
+        mod_usr_tags = usr_tags
+        # Tags shall match when "_" are used instead of "-" or vise versa
+        for character in ["-", "_"]:
+            mod_usr_tags = dict(
+                (key.replace(character, ""), value)
+                for (key, value) in mod_usr_tags.items()
+            )
```
Comment (sebclrsn) [inline src/pykiso/test_coordinator/test_execution.py:113]:
```suggestion
            mod_usr_tags = {
                key.replace(character, ""): value
                for key, value in mod_usr_tags.items()
            }
```

File: src/pykiso/test_coordinator/test_execution.py [hunk 2]
```diff
@@ -98,18 +98,33 @@ def apply_tag_filter(
...
+            test_case.tag = dict(
+                (key.replace(character, ""), value)
+                for (key, value) in test_case.tag.items()
+            )
```
Comment (sebclrsn) [inline src/pykiso/test_coordinator/test_execution.py:117]:
```suggestion
            test_case.tag = {
                key.replace(character, ""): value
                for key, value in test_case.tag.items()
            )
```

**Review by sebclrsn — CHANGES_REQUESTED** [review]

File: src/pykiso/test_coordinator/test_execution.py (hunk 1 above)
Comment (Pog3k) [reply src/pykiso/test_coordinator/test_execution.py]: done

File: src/pykiso/test_coordinator/test_execution.py (hunk 2 above)
Comment (Pog3k) [reply src/pykiso/test_coordinator/test_execution.py]: done

File: src/pykiso/test_coordinator/test_execution.py [hunk 3]
```diff
@@ -98,18 +98,33 @@ def apply_tag_filter(
 
     def is_skip_test(test_case: BasicTest) -> bool:
         """Check if test shall be skipped by evaluating the test case tag
-        attribute
+        attribute.
         :param test_case: test_case to check
         :return: True if test shall be skipped else False
         """
-        for tag_id, tag_value in usr_tags.items():
+        if test_case.tag is None:
```
Comment (sebastianpfischer) [inline src/pykiso/test_coordinator/test_execution.py:105]: This needs to be well unittested (codecov complains)

**Review by sebastianpfischer — CHANGES_REQUESTED** [review]

**Review by sebclrsn — APPROVED** [review]

**Review by sebastianpfischer — APPROVED** [review]

**Pull request merged**